Offline de-compression for string-alike STATIC_URL and COMPRESS_URL

kottenator opened the pull request:
Django doesn't incorporate `SCRIPT_NAME` WSGI param into `settings.STATIC_URL` so I've created `ScriptNamePrefixed(str)` subclass which dynamically adds `SCRIPT_NAME` prefix to itself:

```py
# settings.py
STATIC_URL = ScriptNamePrefixed('/static/')
```

```py
from django.conf import settings
settings.STATIC_URL == '/static/'
# Now WSGI param SCRIPT_NAME is /some/prefix/
settings.STATIC_URL == '/some/prefix/static/'
```

`ScriptNamePrefixed` re-defines all the `str` methods, introducing _the prefix_. However, I can't trick CPython's `str.replace(ScriptNamePrefixed('/static/'), '')` - it ignores _the prefix_ and uses the original string value. But if I replace: 

```py
# compressor/cache.py:55
render_template_string.replace(settings.STATIC_URL, '')
# with
render_template_string.replace(str(settings.STATIC_URL), '')
```
... and:

```py
# compressor/templatetags/compress.py:68
offline_manifest[key].replace(settings.COMPRESS_URL_PLACEHOLDER, settings.COMPRESS_URL)
# with
offline_manifest[key].replace(settings.COMPRESS_URL_PLACEHOLDER, str(settings.COMPRESS_URL))
``` 
&ndash; it works.

This PR adds ability for `settings.STATIC_URL` and `settings.COMPRESS_URL` to be anything that re-defines `__str__` method, not only strings.

diox commented:
It would be best to use `six.text_type` (from `django.utils` to stay consistent with the rest of the code) for this instead of `str`

karyon commented:
i have zero idea what this does and it seems to be quite an edge case, but if @diox says it's okay then it is okay. but i'd suggest adding at least a comment above the two lines, if not tests, to explain why we should cast a setting that's usually a string to a string.

kottenator commented:
@karyon, @diox - does it look good now?

karyon commented:
fine by me, but i'll leave merging to @diox as i'm unsure what this does.

diox commented:
This still needs a test (especially as it's a pretty obscure feature that we're likely to break without realizing)

kottenator commented:
Sure, I'll add it.

karyon commented:
ping :)

kottenator commented:
@karyon - pong :) 
Finally, I have some time to work on my PR. Will add tests today.

kottenator commented:
@karyon, @diox - I've added tests. I've also added some explanation of my motivation. I hope you'll find it useful.

Please, let me know what else can/should I do here ;)

diox commented:
Looks good. The repetition of the comment is a bit un-DRY, but that's fine, we can live with that.

diox commented:
Thank you for your patience :) This is now merged.
